packages/core/addon/components/table-renderer-vertical-collection.js

```diff
@@ -0,0 +1,250 @@
+    }
+
+    if (get(this, 'isEditingMode')) {
+      //reposition real headers for editing mode
+      let translate = `translate(0,${scrollableWrapper.scrollTop - 2}px)`;
+      this.element.querySelectorAll('th').forEach(headerElm => (headerElm.style.transform = translate));
+    }
+  },
+
+  /**
+   * Synchronizes the width of visible table headers with real (hidden) table header elements
+   *
+   * @method _syncHeadersWidth
+   * @private
+   * @returns {void}
+   */
+  _syncHeadersWidth() {
+    //in editing mode the real table headers are visible, no need to sync
+    if (get(this, 'isEditingMode')) {
+      return;
+    }
+
+    let widths = Array.from(this.element.querySelectorAll('th')).map(elm => elm.getBoundingClientRect().width);
+
+    let prevWidths = get(this, 'columnsWidth'),
+      widthChanged = widths.length !== prevWidths.length; //if number of columns has changed, set to true to re-set width
+
+    Array.from(this.element.querySelectorAll('div.table-header-row-vc .table-header-cell')).map((headerElm, i) => {
+      let headerElmWidth = parseInt(headerElm.style.minWidth);
+
+      if (!headerElmWidth || widths[i] !== prevWidths[i]) {
+        headerElm.style.minWidth = `${widths[i]}px`;
+        widthChanged = true;
+      }
+    });
+
+    //set total width to headers wrapper div
+    let headerRowElm = this.element.querySelector('div.table-header-row-vc'),
+      headerRowStyle = parseInt(headerRowElm.style.minWidth);
+
+    if (!headerRowStyle || widthChanged) {
+      set(this, 'columnsWidth', widths);
+      headerRowElm.style.minWidth = `${this.element.querySelector('table').getBoundingClientRect().width}px`;
+    }
+  },
+
+  /**
+   * attach scroll and view resize even listeneres when first rendered
+   *
+   * @method didInsertElement
+   * @override
+   */
+  didInsertElement() {
+    this._super(...arguments);
+
+    run.scheduleOnce('afterRender', () => {
+      if (!get(this, 'isDestroyed') && !get(this, 'isDestroying')) {
+        this._registerTableScroll();
+        this._registerViewResize();
+      }
+    });
+  },
+
+  /**
+   * during both render and re-render, sync headers width and scroll position
+   *
+   * @method didRender
+   * @override
+   */
+  didRender() {
+    this._super(...arguments);
+
+    run.scheduleOnce('render', () => {
+      if (!get(this, 'isDestroyed') && !get(this, 'isDestroying')) {
+        this._syncHeadersWidth();
+        this._syncScroll();
+      }
+    });
+  },
+
+  /**
+   * turn off event listeners before removing from the DOM
+   *
+   * @method willDestroyElement
+   * @override
+   */
+  willDestroyElement() {
+    //turn off scroll listener
+    [get(this, 'tableWrapperDomElement'), get(this, 'tableHeadersDomElement')].forEach(elm =>
+      elm.removeEventListener(SCROLL_EVENT, () => this._syncScroll())
+    );
+
+    //turn off view resize listener
+    get(this, 'resize').off(RESIZE_EVENT);
+
+    this._super(...arguments);
+  },
+
+  actions: {
```

Comment:
Could we just use the `mut` helper to change the value of `isDragged` instead of using these actions?

Reply:
Sure, wasn't familiar with it